homeassistant/components/alarm_control_panel/manual.py

```diff
@@ -86,23 +120,36 @@ class ManualAlarm(alarm.AlarmControlPanel):
     Representation of an alarm status.
 
     When armed, will be pending for 'pending_time', after that armed.
-    When triggered, will be pending for 'trigger_time'. After that will be
-    triggered for 'trigger_time', after that we return to the previous state
-    or disarm if `disarm_after_trigger` is true.
+    When triggered, will be pending for the triggering state's 'delay_time'
+    plus the triggered state's 'pending_time'.
+    After that will be triggered for 'trigger_time', after that we return to
+    the previous state or disarm if `disarm_after_trigger` is true.
+    A trigger_time of zero disables the alarm_trigger service.
     """
 
-    def __init__(self, hass, name, code, pending_time, trigger_time,
+    def __init__(self, hass, name, code, code_template,
+                 delay_time, pending_time, trigger_time,
                  disarm_after_trigger, config):
         """Init the manual alarm panel."""
         self._state = STATE_ALARM_DISARMED
         self._hass = hass
         self._name = name
-        self._code = str(code) if code else None
-        self._trigger_time = datetime.timedelta(seconds=trigger_time)
+        if code_template:
+            self._code = code_template
+            self._code.hass = hass
+        else:
+            self._code = str(code) if code else None
```

Comment:
The code is already validated as a string if set in the config. If we want to check for empty string, I think we should do that already in the config validation. Could you add a validator that raises vol.Invalid for empty string? You should be able to use `vol.IsTrue()`.

Reply:
I think the purpose here is to make `code = ''` equivalent to no code at all. I'll change it to `self._code = code or None`.